src/main/java/me/champeau/jmh/JmhBytecodeGeneratorTask.java

```diff
@@ -53,25 +50,29 @@
     public abstract ConfigurableFileCollection getClassesDirsToProcess();
 
     @OutputDirectory
-    public abstract DirectoryProperty getGeneratedClassesDir();
+    public abstract DirectoryProperty getGeneratedSourcesDir();
 
     @OutputDirectory
-    public abstract DirectoryProperty getGeneratedSourcesDir();
+    public abstract DirectoryProperty getGeneratedResourcesDir();
 
     @TaskAction
     public void generate() {
-        getWorkerExecutor().processIsolation(process -> {
-            if (getJavaLauncher().isPresent()) {
-                process.getForkOptions().executable(getJavaLauncher().get().getExecutablePath().getAsFile());
-            }
-            process.getClasspath().setFrom(getJmhClasspath());
-            process.getForkOptions().jvmArgs(getJvmArgs().get());
-        }).submit(JmhBytecodeGeneratorRunnable.class, params -> {
-            params.getClasspath().from(getRuntimeClasspath());
-            params.getGeneratorType().set(getGeneratorType());
-            params.getOutputSourceDirectory().set(getGeneratedSourcesDir());
-            params.getOutputResourceDirectory().set(getGeneratedClassesDir());
-            params.getClassesDirsToProcess().from(getClassesDirsToProcess());
-        });
+        for (File classesDirsToProcess : getClassesDirsToProcess()) {
```

Comment:
:x: why do you need a loop here? It feels like an error in any case since the classesDirsToProcess is also part of the spec classpath. Shouldn't we be able to do everything in a single round?

Reply:
The program only takes a single directory, so we can't do that. That's why I've also added them to the classpath, so that it can still see all the other classes, in case there are cross-references.

Reply:
We could do a copy operation first to get all classes into one dir and then call the program.

Reply:
Ping :)